Bluetooth: Mesh: HB pub status

Andrewpini opened the pull request:
- Fixes issue where Heartbeat Publication Status message sends garbage data when destination field is set to the unassigned address.
- Fixes incorrect unpacking of Heartbeat Publication status messages on the Configuration client.
- Ensures correct packing of net idx in Heartbeat Publication Status messages on the Configuration Server.

alxelax reviewed (approved):
Try to rebase to pass CI

Andrewpini committed:
Bluetooth: Mesh: HB pub status for unassigned addr

Fixes issue where Heartbeat Publication Status message sends
garbage data when destination field is set to the unassigned address.

MshPRTv1.1, section 4.4.1.2.15:
"When the Destination field is set to the unassigned address, the values
of the CountLog, PeriodLog, TTL, and Features fields shall be set to
0x00 and NetKeyIndex field shall be set to 0x0000."

Signed-off-by: Anders Storrø <anders.storro@nordicsemi.no>

Andrewpini committed:
Bluetooth: Mesh: Correct pack of net idx hb_status

Ensures correct packing of net idx in Heartbeat Publication Status
messages on the Configuration Server.

Reference to net idx packing format: MshPRTv1.1 section 4.3.1.1

Signed-off-by: Anders Storrø <anders.storro@nordicsemi.no>

Andrewpini committed:
Bluetooth: Mesh: Fix unpacking of hb_pub status

Fixes incorrect unpacking of Heartbeat Publication status
messages on the Configuration client.

Reference to packing format: MshPRTv1.1 section 4.3.2.63

Signed-off-by: Anders Storrø <anders.storro@nordicsemi.no>
